CMakeLists.txt

```diff
@@ -23,7 +23,12 @@ if (NOT CMAKE_BUILD_TYPE)
   set(CMAKE_BUILD_TYPE "RelWithDebInfo" CACHE STRING "Build Type" FORCE)
 endif()
 
+include(CMakeDependentOption)
+
 option(BUILD_SHARED_LIBS "Build shared libraries." OFF)
+option(DLAF_WITH_CUDA "Enable CUDA support" OFF)
+cmake_dependent_option(DLAF_WITH_CUDA_MPI_RDMA "Enable MPI CUDA RDMA" OFF "DLAF_WITH_CUDA" OFF)
+cmake_dependent_option(DLAF_WITH_UMPIRE "Enable Umpire support" ON "DLAF_WITH_CUDA" OFF)
```

Comment:
Umpire is mostly useless with the CPU backend. However, it also doesn't seem to hurt, and with spack, having it as a dependency is rather trivial. My preference would be to actually make it a required dependency since it avoids having to run tests with and without it. On the other hand, I see why it might be nice to not force it as a dependency (although for CUDA it's practically a must if one wants good performance).

Preferences?

Reply:
My guess is that DLA-F will be built with CUDA support in production much more often than not. If that is indeed the case, I don't mind making Umpire a required dependency.

Reply:
I'll make it required in that case.